### PR TITLE
Feat fast sync block by height

### LINKED ISSUE #1655: Never Return A Block/BlockHeader With Insufficient Finality Signatures To a Fetcher
When a participating node receives a fetcher request for a `Block` or `BlockHeader` by height, it should never return a block header with insufficient finality signature weight.

If it does, it should be banned.

### PR DESCRIPTION
**Pull request by xcthulhu**
Ensures that we only send 2/3rds finality signatures over the wire when making a fetcher request.

Closes #1655

### CODE REVIEW

File: node/src/components/storage/tests.rs [hunk 1]
```diff
@@ -429,9 +473,24 @@ fn test_get_block_header_and_finality_signatures_by_height() {
         );
     }
 
+    let genesis_validator_weights: BTreeMap<PublicKey, U512> =
+        vec![(alice_public_key, 123.into()), (bob_public_key, 123.into())]
+            .into_iter()
+            .collect();
+    let finality_threshold_fraction = Ratio::new(1, 3);
+    let switch_block =
+        switch_block_for_block_header(block.header(), genesis_validator_weights.clone());
```
Comment (fizyk20) [inline node/src/components/storage/tests.rs:508]: This will panic if the block is in era 0 or at height 0, won't it? And I think `Block::random()` doesn't guarantee that it _won't_ be, so we might get spurious failures. Or am I missing something?

Comment (fizyk20) [reply node/src/components/storage/tests.rs:508]: Ah, nevermind, I see that you've modified `Block::random` to fix this :+1:
Although, I think it might be better to leave `Block::random` the way it was (era 0 or height 0 could be edge cases we might want to test, too) and just make sure in this test that this doesn't happen (for example, by re-generating the block while the era or the height are 0).

**Review by fizyk20 — APPROVED** [review]
Looks good :+1: I only have one nitpick.

File: node/src/components/storage/tests.rs (hunk 1 above)
Comment (xcthulhu) [reply node/src/components/storage/tests.rs:508]: While I agree, we should probably just have explicit tests for those edge cases. I captured this here: https://github.com/casper-network/casper-node/issues/1789
